File: root/cgi-bin/print_env_body.py
#!/usr/bin/env python

import os
import sys
import html

def main():
    print("HTTP/1.1 200 OK")
    print("Content-Type: text/html")
    print("Access-Control-Allow-Origin: *")
    print("Access-Control-Allow-Methods: GET, POST, DELETE")
    print("Access-Control-Allow-Headers: *")

    content_length = os.environ.get('CONTENT_LENGTH')
    if content_length:
        try:
            input_data = sys.stdin.read(int(content_length))
        except (ValueError, IOError):
            input_data = ""
    else:
        input_data = ""
    env_vars = "\n".join(
        [f"<li>{key}={value}</li>" for key, value in os.environ.items()])
    html_content = "<html>\n"
    html_content += "<head>\n"
    html_content += "<title>Environment Variables</title>\n"
    html_content += "</head>\n"
    html_content += "<body>\n"
    html_content += "<h1>Environment Variables</h1>\n"
    html_content += f"<ul>{env_vars}</ul>\n"
    html_content += "<h2>Body</h2>\n"
    html_content += f"<div>{html.escape(input_data)}</div>\n"
    html_content += "</body>\n"
    html_content += "</html>\n"
    # html_content = "<html><body>HELLO WORLD!</body></html>"
    print(f"Content-Length: {len(html_content.encode('utf-8'))}", end="\r\n\r\n")
    print(html_content, end="")

if __name__ == "__main__":
    main()

File: root/cgi-bin/print_env_body.php
#!/usr/bin/env php
<?php
// Get the request body
// $requestBody = file_get_contents("php://input");

// Prevent PHP from sending the default Content-Type header
header("Content-Type: text/html");  // Set custom Content-Type header
header_remove("Content-Type"); 

// $htmlContent = "<html><body>Hello World!</body></html>";

// Get the request body
$requestBody = file_get_contents("php://input");

// Generate HTML content and store it in a variable
$htmlContent = "<html>";
$htmlContent .= "<head><title>Environment Variables</title></head>";
$htmlContent .= "<body>";
$htmlContent .= "<h1>Environment Variables</h1>";
$htmlContent .= "<ul>";

// Iterate through environment variables and display them
foreach ($_SERVER as $key => $value) {
    $htmlContent .= "<li>" . htmlspecialchars($key) . "=" . htmlspecialchars($value) . "</li>";
}

$htmlContent .= "</ul>";
$htmlContent .= "<h2>Body</h2>";
$htmlContent .= "<div>" . htmlspecialchars($requestBody) . "</div>";
$htmlContent .= "</body>";
$htmlContent .= "</html>";

// Calculate the Content-Length
$contentLength = strlen($htmlContent);

// Output HTTP headers with Content-Length
echo("HTTP/1.1 200 OK\n");
echo("Content-Type: text/html\n");
echo("Access-Control-Allow-Origin: *\n");
echo("Access-Control-Allow-Methods: GET, POST, DELETE\n");
echo("Access-Control-Allow-Headers: *\n");
echo("Content-Length: " . $contentLength . "\r\n\r\n");
// Output the HTML content
echo $htmlContent;
?>

